app/filevalueresolver.go

```diff
@@ -0,0 +1,74 @@
+package app
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"github.com/TIBCOSoftware/flogo-lib/config"
+	"github.com/TIBCOSoftware/flogo-lib/logger"
+)
+
+var preload = make(map[string]interface{})
+
+var log = logger.GetLogger("app-props-resolver-file")
+
+// Comma separated list of json files overriding default application property values
+// e.g. FLOGO_APP_PROPS_FILE_CONFIG=app1.json,common.json
+const EnvAppPropertyFileConfigKey = "FLOGO_APP_PROPS_FILE_CONFIG"
```

Comment:
Since we are somehow redesigning the configuration management in Flogo, I wonder if existing prefixes are good.
* **FLOGO_** : OK
* **APP_**: is it really useful ? all other environment variables such as "FLOGO_DATA_SECRET_KEY", "FLOGO_PUBLISH_AUDIT_EVENTS", "FLOGO_LOG_LEVEL"... target a Flogo Application. I think we should drop this from the prefix for configuration mangement for simplicity's sake :)
* However since we are in the configuration management scope, we could add a **CONFIG_** prefix instead
* Lastly, the **PROPS_FILE** is misleading since the expected format is not a standard key=value properties file but a JSON file. It would be better to use something more explicit like **PROFILES**. The documentation would explain what a *profile* is in the Flogo context. It's consistent with other tools like BW for instance and the trailing **S** emphasizes the fact that the variable can contain multiple values (comma-separated list of *profiles*).

To sum up, I would suggest the following environment variable name: **FLOGO_CONFIG_PROFILES**

Reply:
@debovema : I prefer more explicit naming to give developers context of the configuration. In my opinion, FLOGO_CONFIG_PROFILES does not give clear indication of what kind configuration. Is it engine, app props or something else?
I think, we should follow consistent naming like FLOGO_APP_PROPS_FILE_OVERRIDE (for file resolver), FLOGO_APP_PROPS_CONSUL_OVERRIDE (for Consul resolver), FLOGO_APP_PROPS_ENV_OVERRIDE (for env resolver). This kind of naming give me clear indication of what it really does.

Reply:
In addition to @debovema comment, I feel that the FLOGO_CONFIG prefix is locked because  
FLOGO_CONFIG_PATH is used to point the flogo.json that defines the application.
What about renaming FLOGO_CONFIG_PATH to FLOGO_APP_PATH, then use FLOGO_CONFIG prefix for configuration management ?

Reply:
I don't think `FLOGO_APP_PROPS_FILE_OVERRIDE` is great... However, at the same time, overloading the term config may become a bit of a challenge, as the app.json, in itself, is really just a config file for the engine... Inside this config, we have the notion of properties... So in reality, you're using (potentially) an external configuration mgmt tool to override properties within your flogo app config... We surface the notion of application just about everywhere, however, so I suspect differentiating "engine" config vs "app" config may make sense.

@emilienthomas suggestion in renaming `FLOGO_CONFIG_PATH` to `FLOGO_APP_PATH` may make sense... However, I'd still suggest that keeping everything properly scoped is important... So, regardless of that change, you'd still probably want to have something like `FLOGO_APP_CONFIG_PROPERTY_RESOLVER` or something, which begins to get a bit long imo...

Reply:
How about following?
```
FLOGO_APP_PROPS_CONFIG_FILE
FLOGO_APP_PROPS_CONFIG_CONSUL
FLOGO_APP_PROPS_CONFIG_ENV
```

Reply:
@debovema: As part of PR I reused your implementation for env resolver. This will help us find better and consistent naming. :-)